crates/pixi_manifest/src/toml/task.rs

```diff
@@ -71,6 +74,35 @@ impl<'de> toml_span::Deserialize<'de> for TaskArg {
 /// A task defined in the manifest.
 pub type TomlTask = WithWarnings<Task>;
 
+impl<'de> toml_span::Deserialize<'de> for DependencyArg {
+    fn deserialize(value: &mut Value<'de>) -> Result<Self, DeserError> {
+        match value.take() {
+            ValueInner::String(s) => Ok(DependencyArg::Positional(TemplateString::new(
+                s.into_owned(),
+            ))),
+            ValueInner::Table(table) => {
+                if table.len() != 1 {
+                    return Err(DeserError::from(toml_span::Error {
+                        kind: ErrorKind::Custom(
+                            "expected exactly one key in named argument".into(),
+                        ),
+                        span: value.span,
+                        line_info: None,
+                    }));
+                }
+
+                let (k, mut v) = table.into_iter().next().unwrap();
```

Comment:
Instead of `next` and the check above, you can also use https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.exactly_one